docker/user_service-ubuntu.sh

```diff
@@ -2,27 +2,47 @@
 [ -f .bashrc ] && . .bashrc
 set -ev
 INSTALL_DIR="${1:-/opt}"
+if [ -n "${PYTHON_EXECUTABLE}" ]; then 
+  PYTHON=$PYTHON_EXECUTABLE
+else
+  PYTHON='miniconda'
+fi
 
 # SIRF-Exercises
 git clone https://github.com/SyneRBI/SIRF-Exercises --recursive -b master $INSTALL_DIR/SIRF-Exercises
 
-if [ -f requirements-service.txt ]; then
-  conda install -c conda-forge -y --file requirements-service.txt || \
-  pip install -U -r requirements-service.txt
-fi
+if [ "$PYTHON" = "miniconda" ]; then
+  # updates the various packages
+  conda update -c conda-forge -y --all
+  if [ -f requirements-service.yml ]; then
+    # installs the required packages in the environment with requirements-service.yml. 
+    # Notice that these requirements TEMPORARILY contains also the packages for SIRF-Exercises
+    mamba env update --file requirements-service.yml 
+  fi
+  conda clean -y --all
 
-#install SIRF-Exercises requirements
-cd $INSTALL_DIR/SIRF-Exercises
-if [ -f requirements.txt ]; then
-  conda install -c conda-forge -y --file requirements.txt || \
-  pip install -U -r requirements.txt
+# Python (runtime)
+else
+  if [ -f requirements-service.txt ]; then
+    pip install -U -r requirements-service.txt
```

Comment:
maybe this is running in a venv, but I guess this might be safer
```suggestion
    $PYTHON_EXECUTABLE -m pip install -U -r requirements-service.txt
```

Reply:
actually, should this be run in the same virtualenv as in `user_python_ubuntu.sh`?